### PR TITLE
First improvement using cppcheck and google test tools

### PR DESCRIPTION
**Pull request by danimartin82**
**cppcheck** tool is able of detecting potential errors in the code. After running it on raylib/src folder, some errors were found in core.c file in function SaveStorageValue(). 

* In the first commit, the function  SaveStorageValue() has been improved.
* In the second commit **Google Test** tool is added, with a README.md file. Also 5 UT for testing the function SaveStorageValue().

Also tested using the example **core_storage_values.c** 

**Important**: still pending an improvement: what happen with functions `SaveStorageValue(int position, int value)` and `LoadStorageValue(int position)` when `position `is negative?  My proposal here will be to change this argument in header of the function to `unsigned int.`

### CODE REVIEW

File: src/core.c [hunk 1]
```diff
@@ -2208,30 +2208,52 @@ void SaveStorageValue(int position, int value)
 #endif
 
     int dataSize = 0;
+    int new_dataSize = 0;
```
Comment (raysan5) [inline src/core.c]: Hey, I avoid snake_case variables in raylib, please use camelCame (`newDataSize`).

Comment (raysan5) [reply src/core.c]: By the way: https://github.com/raysan5/raylib/wiki/raylib-coding-conventions

**Comment by raysan5** [comment]
@danimartin82 I usually try to avoid additional tools in raylib repo (GoogleTest), simple reason it's because I can not maintain it and it generates new Issues pointing to me, could those tests be moved to a separate repo?

As a side note, right now I'm re-writting all TRACELOG() messages for raylib 3.0 release and I can not merge (I need to upload my changes first).

**Comment by danimartin82** [comment]
oook!!! Thanks for the feedback. :-)  I will reorganize this PR and I will push the UTs in a different repo.

**Comment by raysan5** [comment]
@danimartin82 by the way, I already pushed all tracelog changes.

**Pull request closed without merging**